src/proxy/inbound.rs

```diff
@@ -187,6 +215,22 @@ impl Inbound {
                 }
 
                 let addr: SocketAddr = addr.unwrap();
+                if addr.ip() != conn.dst.ip() {
+                    info!("Sending 400, ip mismatch {addr} != {}", conn.dst);
+                    return Ok(Response::builder()
+                        .status(StatusCode::BAD_REQUEST)
+                        .body(Body::empty())
+                        .unwrap());
+                }
+                // Orig has 15008, swap with the real port
+                let conn = rbac::Connection { dst: addr, ..conn };
+                if !workloads.assert_rbac(&conn).await {
```

Comment:
Just tested locally with the local redirect method and I'm getting:

```
2022-12-28T16:48:39.049473Z  INFO inbound{id=2f63be3786443ac0803b0dd43a307b6a peer_ip=127.0.0.1 peer_id=spiffe://cluster.local/ns/default/sa/default}: ztunnel::proxy::inbound: RBAC rejected conn=127.0.0.1(spiffe://cluster.local/ns/default/sa/default)->127.0.0.1:8000
```

Reply:
`no allow policies matched`

Reply:
do you have policies defined?

Reply:
There is an `allow: not 9999` in local.yaml but I don't think it's matching correctly. See the other comment.